src/app/services/coin.service.ts

```diff
@@ -15,6 +15,7 @@ export class CoinService {
   constructor() {
     this.loadAvailableCoins();
     this.loadCurrentCoin();
+    this.saveCoin(this.currentCoin.getValue().id);
```

Comment:
Please use `sessionStorage.setItem(this.storageKey, this.currentCoin.getValue().id.toString());` instead of calling `saveCoin`, because it changes the selected coin in `localStorage` every time a tab is refreshed (assuming that the user has several tabs with different coins).